### PR TITLE
fix: Bump SQLAlchemy

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

Bumps SQLAlchemy and fixes a few of our model definitions. I'm not really sure how these used to work, since they're clearly wrong.

## Which issue(s) this PR fixes:

_(REQUIRED)_

N/A

### CODE REVIEW

File: mealie/schema/group/group_shopping_list.py [hunk 1]
```diff
@@ -225,10 +225,14 @@ class ShoppingListUpdate(ShoppingListSave):
 
 
 class ShoppingListOut(ShoppingListUpdate):
-    recipe_references: list[ShoppingListRecipeRefOut]
-    label_settings: list[ShoppingListMultiPurposeLabelOut]
+    recipe_references: list[ShoppingListRecipeRefOut] = []
+    label_settings: list[ShoppingListMultiPurposeLabelOut] = []
```
Comment (hay-kot) [inline mealie/schema/group/group_shopping_list.py:229]: I don't think you can use the array like this? Won't it always be a reference to the same array?

File: mealie/db/models/group/shopping_list.py [hunk 2]
```diff
@@ -124,15 +124,15 @@ class ShoppingList(SqlAlchemyBase, BaseMixins):
...
         ShoppingListItem,
         cascade="all, delete, delete-orphan",
         order_by="ShoppingListItem.position",
         collection_class=ordering_list("position"),
     )
 
-    recipe_references: Mapped[ShoppingListRecipeReference] = orm.relationship(
-        ShoppingListRecipeReference, cascade="all, delete, delete-orphan"
+    recipe_references: Mapped[list[ShoppingListRecipeReference]] = orm.relationship(
+        ShoppingListRecipeReference, cascade="all, delete, delete-orphan", uselist=True
```
Comment (hay-kot) [inline mealie/db/models/group/shopping_list.py]: how come uselist=True is added to some but not all of them?

File: mealie/schema/group/group_shopping_list.py (hunk 1 above)
Comment (michael-genson) [reply mealie/schema/group/group_shopping_list.py:229]: Normally that's true, but Pydantic does some special magic and makes it safe: https://stackoverflow.com/a/63808835

File: mealie/db/models/group/shopping_list.py (hunk 2 above)
Comment (michael-genson) [reply mealie/db/models/group/shopping_list.py]: Let me remove those, originally I thought it made a difference but that was before I noticed the mapped attribute type was wrong. It's unnecessary when using `Mapped` on a collection: https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.uselist

File: mealie/schema/group/group_shopping_list.py (hunk 1 above)
Comment (hay-kot) [reply mealie/schema/group/group_shopping_list.py:229]: Thought this was in the models folder, ignore me! :D

**Review by boc-the-git — APPROVED** [review]
LGTM

**Pull request merged**